File: backend/src/main/java/com/example/cloud/entity/Signalement.java
package com.example.cloud.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "signalements")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Signalement {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(columnDefinition = "TEXT")
    private String description;
    
    @Column(nullable = false)
    private Double latitude;
    
    @Column(nullable = false)
    private Double longitude;
    
    @Column(length = 20)
    @Builder.Default
    private String status = "NOUVEAU"; // 'NOUVEAU', 'EN_COURS', 'TERMINE'
    
    @Column(name = "surface_m2", precision = 10, scale = 2)
    private BigDecimal surfaceM2;
    
    @Column(precision = 15, scale = 2)
    private BigDecimal budget;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_entreprise")
    private Entreprise entreprise;
    
    @Column(name = "date_signalement", updatable = false)
    @Builder.Default
    private LocalDateTime dateSignalement = LocalDateTime.now();
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;
}

File: frontend/src/types/signalement.ts
export type SignalementResponse = {
  id: number
  description?: string | null
  latitude: number
  longitude: number
  status: string
  surfaceM2?: number | null
  budget?: number | null
  entrepriseId?: number | null
  entrepriseNom?: string | null
  dateSignalement?: string | null
  dateModification?: string | null
  userId?: number | null
  userEmail?: string | null
}

export type SignalementCreateRequest = {
  description?: string | null
  latitude: number
  longitude: number
  surfaceM2?: number | null
  budget?: number | null
  entrepriseId?: number | null
  userId: number
}

export type SignalementUpdateRequest = {
  description?: string | null
  latitude?: number
  longitude?: number
  status?: string
  surfaceM2?: number | null
  budget?: number | null
  entrepriseId?: number | null
}
